theme/static/js/hs-vue/left-header-app.js

```diff
@@ -555,15 +562,22 @@ let leftHeaderApp = new Vue({
     },
     filters: {
         nameWithoutCommas: function (name) {
-            if (!name) return '';
-            name = name.toString();
+            if (!name) return null;
+            name = name.toString().trim();
 
```

Comment:
there is possible type coercion if name is '' (empty string). Returning null or trimmed string is offering returns of variable types. Since you've tested working, maybe just leave this alone as outside Discover code we don't adhere to these coding standards. Just noting for future dev, when frontend code is more consistent.

Reply:
In Python, we are generally more flexible with this and is one of the powerful aspects of the language. In modern JS, with strong typing it's a more disciplined approach that may remind you of other languages you've coded in

Reply:
Good point on type coercion. I know I have a reason to change it to return null if name is empty, potentially related to how null and empty string is interpreted in the author name display, but I am not 100% sure now why I made this change. It does work, so I'll keep it as is, but thanks for the type coercion point which I did not pay attention initially.